### PR TITLE
Added dataframe modify test

### PR DESCRIPTION
**Pull request by jonvanausdeln**
### Intent

Added tests to cover #4281 and #2579 , where modifying a dataframe updates the already open Data Explorer.

 ### Approach

* Added new test to existing dataexplorer suite
* Updated `positronNotebook` to allow for opening an existing notebook file.  The other methods provided by microsoft classes didn't work. Note, I was going to make this extend the Microsoft notebook class, but that class, but I would have to modify thier class to make that possible and didn't think that was worth the trouble.
* Update `positronVariables` to account for more than one variables env being available, by adding z-order

### QA Notes

All smoke tests should pass in CI

### CODE REVIEW

File: test/automation/src/positron/positronNotebooks.ts [hunk 1]
```diff
@@ -23,6 +24,7 @@ const PYTHON_OUTPUT = '.output-plaintext';
 const R_OUTPUT = '.output_container .output';
 const REVERT_AND_CLOSE = 'workbench.action.revertAndCloseActiveEditor';
 const MARKDOWN_TEXT = '#preview';
+const activeRowSelector = `.notebook-editor .monaco-list-row.focused`;
```
Comment (testlabauto) [inline test/automation/src/positron/positronNotebooks.ts:27]: Any reason why you didn't stick with all caps for this one?

File: test/smoke/src/areas/positron/dataexplorer/dataexplorer.test.ts [hunk 2]
```diff
@@ -142,6 +142,51 @@ df2 = pd.DataFrame(data)`;
...
+				await app.workbench.positronConsole.waitForConsoleContents((contents) => contents.some((line) => line.includes('restarted')));
+
+				await app.workbench.positronNotebooks.openNotebook(join(app.workspacePathOrFolder, 'workspaces', 'data-explorer-update-datasets', 'pandas-update-dataframe.ipynb'));
+				await app.workbench.notebook.focusFirstCell();
+				await app.workbench.notebook.executeActiveCell();
+				await expect(async () => {
+					await app.workbench.positronVariables.doubleClickVariableRow('df');
+					await app.code.driver.getLocator('.label-name:has-text("Data: df")').innerText();
+				}).toPass();
+
```
Comment (testlabauto) [inline test/smoke/src/areas/positron/dataexplorer/dataexplorer.test.ts:164]: might want a timeout here

**Review by testlabauto — DISMISSED** [review]
LGTM!  Just had a couple minor comments.

File: test/automation/src/positron/positronNotebooks.ts (hunk 1 above)
Comment (jonvanausdeln) [reply test/automation/src/positron/positronNotebooks.ts:27]: Yes, I blindly copied and pasted it, LOL.

File: test/smoke/src/areas/positron/dataexplorer/dataexplorer.test.ts (hunk 2 above)
Comment (jonvanausdeln) [reply test/smoke/src/areas/positron/dataexplorer/dataexplorer.test.ts:164]: Good catch!

**Pull request merged**